x-pack/plugin/esql-core/src/main/java/org/elasticsearch/xpack/esql/core/expression/FieldAttribute.java

```diff
@@ -145,22 +144,53 @@ public static FieldAttribute readFrom(StreamInput in) throws IOException {
         return ((PlanStreamInput) in).readAttributeWithCache(FieldAttribute::new);
     }
 
+    private void writeParentName(StreamOutput out) throws IOException {
+        if (out.getTransportVersion().onOrAfter(TransportVersions.ESQL_FIELD_ATTRIBUTE_PARENT_SIMPLIFIED)) {
+            out.writeOptionalString(parentName);
+        } else {
+            if (parentName() == null) {
+                out.writeOptionalWriteable((FieldAttribute) null);
+            } else {
+                // Previous versions only used the parent field attribute to retrieve the parent's name, so we can use just any
+                // fake FieldAttribute here.
+                FieldAttribute fakeParent = new FieldAttribute(Source.EMPTY, parentName(), field());
```

Comment:
How about `FieldAttribute fakeParent = parentName()==null ? null : new FieldAttribute(Source.EMPTY, parentName(), field());`, then we can drop the `if` above.

Reply:
Oh, yeah, that's much nicer. Thanks!